mainapp/helpers.py

```diff
@@ -112,7 +112,7 @@ def genre_wise(genre, percentile=0.85):
     qualified = qualified.assign(weighted_rating=W)
     qualified.sort_values('weighted_rating', ascending=False, inplace=True)
 
-    return qualified[cols].head(min_genre_book_count).sample(n_books)
+    return qualified[cols].head(min_genre_book_count).sample(2*n_books)
```

Comment:
If you modify this, then `2*n` goes to genre too, pass `2*n` from where you need itself